### PR TITLE
quilt: fix Ventura build

### PR DESCRIPTION
**Pull request by triallax**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/quilt.rb [hunk 1]
```diff
@@ -31,6 +36,10 @@ def install
     if OS.mac?
       args << "--with-sed=#{HOMEBREW_PREFIX}/bin/gsed"
       args << "--with-stat=/usr/bin/stat" # on macOS, quilt expects BSD stat
+      if MacOS.version >= :ventura
+        args << "--with-diff=#{HOMEBREW_PREFIX}/bin/diff"
+        args << "--with-patch=#{HOMEBREW_PREFIX}/bin/patch"
```
Comment (SMillerDev) [inline Formula/quilt.rb:41]: This doesn't work if not linked. `#{Formula["diffutils"].opt_bin}` would be a better location.

Comment (mhmdanas) [reply Formula/quilt.rb:41]: Doesn't that go for `patch` and `gsed` too?

Comment (SMillerDev) [reply Formula/quilt.rb:41]: Yes it does

**Comment by fxcoudert** [comment]
Why doesn't Ventura `patch` and `diff` work? What's different from Monterey?

**Comment by mhmdanas** [comment]
> Why doesn't Ventura patch and diff work? What's different from Monterey?

I don't know, but the build does error without this PR's changes. Here's the error I get with the Quilt formula on `master`:

```
checking the version of /usr/bin/diff... no GNU diff
configure: error:
quilt requires at least version 2.7 of GNU diffutils. You can
download a current version of patch from ftp.gnu.org, or if you already
have GNU diff then you can supply its path with the '--with-diff=' option.
```

If this error is fixed and you try to build again, a similar error pops up for `patch` too.

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**